File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int i=0;
        int j = 0;
        Set<Character> set = new HashSet<Character>();
        int res = 0;
        while(j<s.length()){
            if(!set.contains(s.charAt(j))){
                set.add(s.charAt(j));
                j++;
            }
            else{
                set.remove(s.charAt(i));
                i++;
            }
            res = Math.max(res,set.size());
        }
        return res;
    }
}

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let set = new Set()
    let i = 0;
    let j = 0;
    let res = 0;
    while(j<s.length){
        if(set.has(s[j])){
            set.delete(s[i])
            i++
        }
        else{
            set.add(s[j])
            res = Math.max(res,set.size)
            j++
        }
    }
    return res
};